features/locale/handling_different_locales_on_multiple_channels.feature

```diff
@@ -1,33 +1,33 @@
 @locales
 Feature: Handling different locales on multiple channels
-    In order to see right prices
+    In order to present shop data in a correct language
     As a Customer
-    I want to browse channels with a valid locale only
+    I want to browse channels with a valid locale
 
     Background:
-        Given the store operates on a channel named "Web"
+        Given the store operates on a channel named "Web" with hostname "web.example"
         And that channel allows to shop using "English (United States)", "Polish (Poland)" and "Norwegian (Norway)" locales
         And it uses the "English (United States)" locale by default
-        And the store operates on another channel named "Mobile"
+        And the store operates on another channel named "Mobile" with hostname "mobile.example"
         And that channel allows to shop using "Polish (Poland)" and "Norwegian (Norway)" locales
         And it uses the "Polish (Poland)" locale by default
 
     @ui
     Scenario: Showing locales only from the current channel
         When I browse the "Mobile" channel
-        Then I should shop using the "polski (Polska)" locale
+        Then I should shop using the "Polish (Poland)" locale
         And I should be able to shop using the "norweski (Norwegia)" locale
         And I should not be able to shop using the "angielski (Stany Zjednoczone)" locale
 
-    @ui
+    @ui @api
     Scenario: Browsing channels using their default locales
-        When I browse the "Web" channel
+        Given I browse the "Web" channel
         And I start browsing the "Mobile" channel
-        Then I should shop using the "polski (Polska)" locale
+        Then I should shop using the "Polish (Poland)" locale
```

Comment:
Here I use a new transformer `LocaleContext::castToItsLocaleName`. I had a problem that UI expects locale names in polish while in API I check the code.

Reply:
It is handled now by `:localeNameInCurrentLocale`.